### PR TITLE
feat: add plugins on Optimize ES

### PR DESCRIPTION
**Pull request by igpetrov**
## Description

feat: add plugins on Optimize ES

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

Related to / built on top of https://github.com/camunda/camunda/pull/22299

### CODE REVIEW

File: optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/db/ElasticsearchDatabaseTestService.java
Comment (igpetrov) [inline optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/db/ElasticsearchDatabaseTestService.java:1]: It seems like zeebe autoformatter / spotless plugin changes order of method. I only added `PluginRepository` here

File: optimize/util/optimize-commons/src/main/java/io/camunda/optimize/service/db/es/OptimizeElasticsearchClient.java
Comment (igpetrov) [inline optimize/util/optimize-commons/src/main/java/io/camunda/optimize/service/db/es/OptimizeElasticsearchClient.java:1]: Same here, autoformatter did a lot of formatting

File: optimize/util/optimize-commons/src/main/java/io/camunda/optimize/upgrade/es/ElasticsearchClientBuilder.java [hunk 1]
```diff
@@ -60,10 +67,15 @@ public static String getCurrentESVersion(
     return esClient.withTransportOptions(requestOptions).info().version().number();
   }
 
-  public static RestClient restClient(final ConfigurationService configurationService) {
+  public static RestClient restClient(
+      final ConfigurationService configurationService, final PluginRepository pluginRepository) {
+    final var plugins = extractPluginConfigs(configurationService.getElasticSearchConfiguration());
```
Comment (oleksandr-kriuchenko-lohika) [inline optimize/util/optimize-commons/src/main/java/io/camunda/optimize/upgrade/es/ElasticsearchClientBuilder.java:72]: Our convention is to not use `var` in the project to my knowledge

**Review by oleksandr-kriuchenko-lohika — APPROVED** [review]

File: optimize/util/optimize-commons/src/main/java/io/camunda/optimize/upgrade/es/ElasticsearchClientBuilder.java (hunk 1 above)
Comment (igpetrov) [reply optimize/util/optimize-commons/src/main/java/io/camunda/optimize/upgrade/es/ElasticsearchClientBuilder.java:72]: Sure, will fix that

**Pull request merged**